src/main/java/org/broadinstitute/hellbender/utils/bwa/BwaMemIndex.java

```diff
@@ -0,0 +1,182 @@
+        try ( final FileInputStream is = new FileInputStream(fileName) ) {
+            return is.read() != -1;
+        } catch ( final IOException ioe ) {
+            return false;
+        }
+    }
+
+    private static void loadNativeLibrary() {
+        if ( !nativeLibLoaded ) {
+            synchronized(BwaMemIndex.class) {
+                if ( !nativeLibLoaded ) {
+                    final String libNameOverride = System.getProperty("LIBBWA_PATH");
+                    if ( libNameOverride != null ) {
+                        System.load(libNameOverride);
+                    }
+                    else {
+                        final String osName = System.getProperty("os.name", "unknown").toUpperCase();
+                        final String osArch = System.getProperty("os.arch");
+                        final String libName;
+                        if ( !"x86_64".equals(osArch) && !"amd64".equals(osArch) ) libName = null;
+                        else if ( osName.startsWith("MAC") ) libName = "/libbwa.Darwin.dylib";
+                        else if ( osName.startsWith("LINUX") ) libName = "/libbwa.Linux.so";
+                        else libName = null;
+                        if ( libName == null ) {
+                            throw new IllegalStateException("We have a JNI binding for bwa-mem only for x86-64 Linux and Mac.");
+                        }
+                        try ( final InputStream is = BwaMemIndex.class.getResourceAsStream(libName) ) {
+                            if ( is == null ) {
+                                throw new IllegalStateException("Can't find resource "+libName);
+                            }
+                            final File tmpFile = File.createTempFile("libbwa.",".jnilib");
+                            tmpFile.deleteOnExit();
+                            Files.copy(is, tmpFile.toPath(), StandardCopyOption.REPLACE_EXISTING);
+                            System.load(tmpFile.getPath());
+                        }
+                        catch (IOException ioe ) {
+                            throw new IllegalStateException("Misconfiguration: Unable to load bwa-mem native library "+libName, ioe);
+                        }
+                    }
+                    nativeLibLoaded = true;
+                }
+            }
+        }
+    }
+
+    private static native boolean createIndexImageFile( String referenceName, String imageName );
+    private static native long openIndex( String indexImageFile );
+    private static native int destroyIndex( long indexAddress );
+    static native ByteBuffer createDefaultOptions();
+    private static native ByteBuffer getRefContigNames( long indexAddress );
+    private static native ByteBuffer createAlignments( ByteBuffer seqs, long indexAddress, ByteBuffer opts );
+    static native void destroyByteBuffer( ByteBuffer alignments );
+    public static native String getVersion();
```

Comment:
It seems like this should be private and `getBWAVersion()` should be the public one.